src/Swarm/Language/LSP/Hover.hs

```diff
@@ -116,10 +138,21 @@ treeToMarkdown :: Int -> Tree (DocLine Text) -> Text
 treeToMarkdown d (Node (DocLine _n t) children) =
   T.unlines $ renderDoc (DocLine d t) : map (treeToMarkdown $ d + 1) children
 
-explain :: Term -> Tree (DocLine Text)
-explain = \case
+-- | Top level explanations can use fancy markdown syntax, but list items
+--   have to be more conservative and ideally shorter.
+data Length = Brief | Long
+  deriving (Eq, Show)
+
+explain :: Length -> Maybe TModule -> Term -> Tree (DocLine Text)
+explain len mMod trm = case trm of
   TUnit -> pure $ pureDoc "The unit value."
-  TConst c -> pure $ pureDoc $ briefDoc $ constDoc $ constInfo c
+  TConst c ->
+    pure . pureDoc $
+      typeSignature
+        len
+        (prettyText c)
+        (Just $ inferConst c)
+        (briefDoc $ constDoc $ constInfo c)
   TDir {} -> pure $ pureDoc "A direction literal."
```

Comment:
While we're at it, could we display something like `left : dir` rather than `A direction literal`?

Reply:
I would show both. 🙂